src/components/Notifications/NotificationsLog/NotificationsLogDateFilter.tsx

```diff
@@ -0,0 +1,295 @@
+import {
+  Bullseye,
+  DatePicker,
+  DatePickerRef,
+  Select,
+  SelectOption,
+  SelectOptionObject,
+  SelectVariant,
+  Split,
+  SplitItem,
+  TextInputProps,
+} from '@patternfly/react-core';
+import {
+  global_active_color_100,
+  global_palette_black_600,
+  global_spacer_sm,
+} from '@patternfly/react-tokens';
+import { important } from 'csx';
+import { add, format, isAfter, isBefore, min, parseISO } from 'date-fns';
+import produce from 'immer';
+import * as React from 'react';
+import { Dispatch, useRef } from 'react';
+import { SetStateAction } from 'react';
+import { style } from 'typestyle';
+
+import { EventPeriod } from '../../../types/Event';
+
+export enum NotificationsLogDateFilterValue {
+  TODAY = 'today',
+  YESTERDAY = 'yesterday',
+  LAST_7 = 'last_7',
+  LAST_14 = 'last_14',
+  CUSTOM = 'custom',
+}
+
+const labels: Record<NotificationsLogDateFilterValue, string> = {
+  [NotificationsLogDateFilterValue.TODAY]: 'Today',
+  [NotificationsLogDateFilterValue.YESTERDAY]: 'Yesterday',
+  [NotificationsLogDateFilterValue.LAST_7]: 'Last 7 days',
+  [NotificationsLogDateFilterValue.LAST_14]: 'Last 14 days',
+  [NotificationsLogDateFilterValue.CUSTOM]: 'Custom',
+};
+
+const toClassName = style({
+  paddingLeft: global_spacer_sm.value,
+  paddingRight: global_spacer_sm.value,
+  color: global_palette_black_600.value,
+});
+
+const datePickerClassName = style({
+  backgroundColor: important('white'),
+  cursor: 'pointer',
+  $nest: {
+    '&::placeholder': {
+      color: important('black'),
+    },
+    '&:hover': {
+      borderBottomColor: global_active_color_100.value,
+    },
+  },
+});
+
+class EventLogSelectObject implements SelectOptionObject {
+  readonly value: NotificationsLogDateFilterValue;
+
+  constructor(value: NotificationsLogDateFilterValue) {
+    this.value = value;
+  }
+
+  toString(): string {
+    return labels[this.value];
+  }
+  compareTo(selectOption: unknown): boolean {
+    if (selectOption instanceof EventLogSelectObject) {
+      return selectOption.value === this.value;
+    }
+
+    return false;
+  }
+}
+
+const dateInputProps: TextInputProps = {
+  isReadOnly: true,
+  className: datePickerClassName,
+};
+
+interface CustomDateFilterProps {
+  retentionDays: number;
+  period: EventPeriod;
+  setPeriod: Dispatch<SetStateAction<EventPeriod>>;
+}
+
+const CustomDateFilter: React.FunctionComponent<CustomDateFilterProps> = (
+  props
```

Comment:
You should break the props, it's more readable that way.
```suggestion
  {
    period,
    setPeriod
  }
```

Reply:
Fixed

Reply:
Fixed